### PR TITLE
Add descriptions for more HTTP status codes

### LINKED ISSUE #2871: [Feature request]: Add descriptions for more HTTP status codes
### Is your feature request related to a specific problem? Or an existing feature?

I've noticed that the HTTP status code 422 has the description "Client Error" by default. I'd expect it to be "Unprocessable Entity".
Seems like "Client Error" is the default description for all 4xx HTTP status codes that don't have a better description.

### Describe the solution you'd like

Please update the default HTTP status code description list. A good source seems to be https://github.com/dotnet/aspnetcore/blob/main/src/Http/WebUtilities/src/ReasonPhrases.cs

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by cremor**
# Pull Request

## The issue or feature being addressed
Fixes #2871

## Details on the issue fix or feature implementation
As suggested in the issue, I've mostly added the status codes missing when compared with https://www.rfc-editor.org/rfc/rfc9110.html#name-status-codes

The following status codes are not in that RFC but I've added them because I think they make sense. You can read details about them here: https://httpstatuses.io

423 Locked
424 Failed Dependency
428 Precondition Required
429 Too Many Requests (was already there)
431 Request Header Fields Too Large
451 Unavailable For Legal Reasons

I hope the readme changes for 200 OK and 500 Internal Server Error are ok. If not, please tell if I should only revert the readme changes, or if I should revert the whole 200 OK and 500 Internal Server Error lines (so that they fall back to the generic description like before).

### CODE REVIEW

File: src/Swashbuckle.AspNetCore.SwaggerGen/SwaggerGenerator/SwaggerGenerator.cs [hunk 1]
```diff
@@ -667,31 +667,64 @@ private static bool IsFromFormAttributeUsedWithIFormFile(ApiParameterDescription
...
             new KeyValuePair<string, string>("404", "Not Found"),
             new KeyValuePair<string, string>("405", "Method Not Allowed"),
             new KeyValuePair<string, string>("406", "Not Acceptable"),
+            new KeyValuePair<string, string>("407", "Proxy Authentication Required"),
             new KeyValuePair<string, string>("408", "Request Timeout"),
             new KeyValuePair<string, string>("409", "Conflict"),
+            new KeyValuePair<string, string>("410", "Gone"),
+            new KeyValuePair<string, string>("411", "Length Required"),
+            new KeyValuePair<string, string>("412", "Precondition Failed"),
+            new KeyValuePair<string, string>("413", "Payload Too Large"),
```
Comment (martincostello) [inline src/Swashbuckle.AspNetCore.SwaggerGen/SwaggerGenerator/SwaggerGenerator.cs:706]:
```suggestion
            new KeyValuePair<string, string>("413", "Content Too Large"),
```

**Review by martincostello — APPROVED** [review]

**Comment by martincostello** [comment]
Looks like something in the tests is broken or needs updating.

File: src/Swashbuckle.AspNetCore.SwaggerGen/SwaggerGenerator/SwaggerGenerator.cs (hunk 1 above)
Comment (cremor) [reply src/Swashbuckle.AspNetCore.SwaggerGen/SwaggerGenerator/SwaggerGenerator.cs]: Thanks, I missed that in RFC9110. Should we do the same for 422 Unprocessable Content? (I've added it as Unprocessable Entity.)

Comment (martincostello) [reply src/Swashbuckle.AspNetCore.SwaggerGen/SwaggerGenerator/SwaggerGenerator.cs]: Good spot - I missed that one.

**Comment by cremor** [comment]
Tests should be fixed.

File: src/Swashbuckle.AspNetCore.SwaggerGen/SwaggerGenerator/SwaggerGenerator.cs (hunk 1 above)
Comment (cremor) [reply src/Swashbuckle.AspNetCore.SwaggerGen/SwaggerGenerator/SwaggerGenerator.cs]: Fixed that too.

**Pull request merged**